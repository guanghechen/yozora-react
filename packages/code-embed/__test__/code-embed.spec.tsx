import CodeRendererJsx from '@yozora/react-code-renderer-jsx'
import { render } from 'enzyme'
import React from 'react'
import type { CodeRunnerProps } from '../src'
import CodeEmbed from '../src'

const code = `
  function Counter() {
    const [count, setCount] = React.useState(0)
    return (
      <div>
        <button onClick={() => setCount(c => c + 1)}>+</button>
        <span>{count}</span>
        <button onClick={() => setCount(c => c - 1)}>-</button>
      </div>
    )
  }
`

const JsxRunner = ({ value }: CodeRunnerProps): React.ReactElement => {
  return (
    <CodeRendererJsx
      code={value}
      inline={true}
      onError={error => {
        console.log(error)
      }}
    />
  )
}

describe('prop types', function () {
  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })
  })

  it('value is required', function () {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<CodeEmbed lang="jsx" value={value} CodeRunner={JsxRunner} />)
      }).toThrow(/The prop `value` is marked as required/i)
    }
  })

  describe('className is optional', function () {
    it('default', function () {
      const node = render(
        <CodeEmbed lang="jsx" value={code} CodeRunner={JsxRunner} />,
      )
      expect(node.hasClass('yozora-code-embed')).toEqual(true)
    })

    it('custom', function () {
      const node = render(
        <CodeEmbed
          lang="jsx"
          value={code}
          CodeRunner={JsxRunner}
          className="my-code-embed"
        />,
      )
      expect(node.hasClass('yozora-code-embed')).toEqual(true)
      expect(node.hasClass('my-code-embed')).toEqual(true)
    })
  })

  it('style is optional', function () {
    const node = render(
      <CodeEmbed
        lang="jsx"
        value={code}
        CodeRunner={JsxRunner}
        style={{ color: 'orange' }}
      />,
    )
    expect(node.css('color')).toEqual('orange')
  })
})

describe('snapshot', function () {
  it('basic', () => {
    const wrapper = render(
      <CodeEmbed lang="jsx" value={code} CodeRunner={JsxRunner} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
