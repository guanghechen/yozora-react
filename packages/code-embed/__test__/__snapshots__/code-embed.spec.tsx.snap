// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c1 {
  position: absolute;
  top: 0;
  left: 0;
  display: none;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}

.c0 {
  position: relative;
  border: none;
  padding: 0;
  background: transparent;
  color: inherit;
}

<div
  class="c0"
>
  <span
    data-type="jsx"
  >
    9
  </span>
  <div
    class="c1"
    hidden=""
  />
</div>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c1 {
  position: absolute;
  top: 0;
  left: 0;
  display: none;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}

.c0 {
  position: relative;
  border: none;
  padding: 2px;
  background: #fff;
  color: #ccc;
}

<Component
  theme={
    Object {
      "yozora": Object {
        "codeEmbed": Object {
          "background": "#fff",
          "border": "none",
          "color": "#ccc",
          "errorBackground": "red",
          "errorColor": "#f8f8f2",
          "errorFontSize": "0.9em",
          "padding": "2px",
        },
      },
    }
  }
>
  <YozoraCodeEmbed
    CodeRenderer={[Function]}
    lang="jsx"
    value="
      const a = 1 + 2;
      return a * a
    "
  >
    <styled.div
      $hasError={false}
      theme={
        Object {
          "yozora": Object {
            "codeEmbed": Object {
              "background": "transparent",
              "border": "none",
              "color": "inherit",
              "errorBackground": "#ff5555",
              "errorColor": "#f8f8f2",
              "errorFontFamily": "Consolas, \\"Source Code Pro\\", monospace, sans-serif",
              "errorFontSize": "0.9em",
              "padding": "0",
            },
          },
        }
      }
    >
      <div
        className="c0"
      >
        <JsxRenderer
          lang="jsx"
          onError={[Function]}
          value="
      const a = 1 + 2;
      return a * a
    "
        >
          <span
            data-type="jsx"
          >
            9
          </span>
        </JsxRenderer>
        <styled.div
          hidden={true}
        >
          <div
            className="c1"
            hidden={true}
          >
            <YozoraCodeEmbedError
              error={null}
            />
          </div>
        </styled.div>
      </div>
    </styled.div>
  </YozoraCodeEmbed>
</Component>
`;
